src/js/techreport/combobox.js

```diff
@@ -0,0 +1,261 @@
+class ComboBox {
+  constructor(element, data, selected) {
+    this.element = element;
+
+    this.data = data;
+    this.filteredData = data;
+    this.focusedOption = -1;
+    this.focusedOptionStr = '';
+    this.selected = selected || [];
+    this.maxSelected = 10;
+
+    this.updateContent();
+    this.monitorInput();
+
+    this.selected.forEach(name => this.showSelectedElement(name));
+    this.element.querySelector('input[type="text"][disabled]').removeAttribute('disabled');
+  }
+
+  updateContent(data) {
+    const rows = data || this.data;
+    // Add options in the dropdown list
+    const listbox = this.element.querySelector('[role="listbox"]');
+    rows.forEach((row, index) => {
+      const option = document.createElement('div');
+      option.setAttribute('role', 'option');
+      option.setAttribute('aria-selected', 'false');
+      option.dataset.name = row;
+      option.dataset.key = index;
+      option.textContent = row;
+      option.id = `${this.element.dataset.id}-${row}`;
```

Comment:
Need to strip out spaces. And maybe other characters?

```suggestion
      option.id = `${this.element.dataset.id}-${row.replaceAll(' ','-')}`;
```